### PR TITLE
MAISTRA-1523 Install some control plane components in parallel

### PR DESCRIPTION
**Pull request by luksa**
The "security" and "galley" have a circular dependency between each other.
It makes sense to deploy them at the same time. Also, deploying some
components in parallel will speed up the installation process.

### CODE REVIEW

**Comment by rcernich** [comment]
This looks okay, but I think it would be better to defer installation of istio.io resources until after galley is up and running.  I think `security` needs to be broken up into `citadel` and `mtls`, as it's the mtls bit which causes the problems.  Ideally, any component which installs istio.io resources should be installed after galley is up and running to avoid issues.

File: pkg/controller/servicemesh/controlplane/reconciler.go [hunk 1]
```diff
@@ -32,31 +32,27 @@ import (
...
-	"istio/charts/galley",
-	"istio/charts/mixer",
-	"istio/charts/pilot",
-	"istio/charts/gateways",
-	"istio/charts/sidecarInjectorWebhook",
-	"istio/charts/grafana",
-	"istio/charts/kiali",
+var orderedCharts = [][]string{
+	{"istio"}, // core istio resources
+	{"istio/charts/security", "istio/charts/galley"}, // both are deployed at the same time because each is a dependency of the other
```
Comment (dgn) [inline pkg/controller/servicemesh/controlplane/reconciler.go]: I agree with @rcernich that this one could be problematic. If the webhookconfig from `galley` is already present when the istio.io resources from `security` are deployed, but the `galley` pod isn't ready yet, we're going to see errors

Comment (dgn) [reply pkg/controller/servicemesh/controlplane/reconciler.go]: I just realized that from 1.6 onwards upstream does not deploy istio.io configuration on installation anymore, so this problem will likely disappear. I suggest we postpone the merge of security and galley until then

Comment (rcernich) [reply pkg/controller/servicemesh/controlplane/reconciler.go]: We'll still need this for backward compatibility.

**Comment by luksa** [comment]
> This looks okay, but I think it would be better to defer installation of istio.io resources until after galley is up and running. I think `security` needs to be broken up into `citadel` and `mtls`, as it's the mtls bit which causes the problems. Ideally, any component which installs istio.io resources should be installed after galley is up and running to avoid issues.

Yeah, I agree. I've reverted the part where the security and galley are installed at the same time so that they are now installed as they were before. I've left the other charts to be installed in parallel.

**Review by dgn — APPROVED** [review]

**Review by rcernich — APPROVED** [review]
lgtm

I assume this does not address the issue with security and galley, where security install would fail if validating webhook was installed, but galley deployment failed to come up (okay, because this was an existing issue; just want to confirm)

**Comment by luksa** [comment]
> lgtm
> 
> I assume this does not address the issue with security and galley, where security install would fail if validating webhook was installed, but galley deployment failed to come up (okay, because this was an existing issue; just want to confirm)

Yup, this needs to be addressed in a different PR. Or not, since we're switching to istiod anyway.

**Pull request merged**